hermes/bin/src/ipfs/task.rs

```diff
@@ -283,6 +306,91 @@ fn topic_message_handler(
     }
 }
 
+/// Handler function for topic message streams (Doc Sync).
+#[allow(
+    clippy::needless_pass_by_value,
+    reason = "The event will be eventually consumed in the handler"
+)]
+fn doc_sync_topic_message_handler(
+    message: hermes_ipfs::rust_ipfs::GossipsubMessage,
+    topic: String,
+) {
+    let Some(ipfs) = HERMES_IPFS.get() else {
+        tracing::error!("Failed to send on_new_doc event. IPFS is uninitialized");
+        return;
+    };
+
+    // TODO: match the topic against a static list.
+    let Some(channel_name) = topic.strip_suffix(".new") else {
+        tracing::error!("Handling an IPFS message on a wrong channel.");
+        return;
+    };
+
+    let payload = match minicbor::decode::<payload::New>(&message.data) {
+        Ok(payload) => DocumentDisseminationBody::from(payload),
+        Err(err) => {
+            tracing::error!(%channel_name, %err, "Failed to decode .new payload from IPFS message");
```

Comment:
Not sure if this should be `error`. Could someone spam our logs by intentionally sending malformed payloads?